### ISSUE TITLE
Change `url.query` requirement level from recommended to conditionally_required (to match `url.path`)

### ISSUE THREAD

**Issue by trask** [open-telemetry/semantic-conventions#112]
Spinning this out from #109 as a separate issue.

Summarizing the discussion:

Pro: this seems like a bug that slipped into https://github.com/open-telemetry/opentelemetry-specification/pull/3355
Con: query is particularly prone to containing sensitive information

**Comment by trask** [comment]
from @Oberon00's https://github.com/open-telemetry/semantic-conventions/pull/109#issuecomment-1590688151

> it seems OK to be "just" recommended. I think that would allow instrumentations to provide an opt-out option which REQUIRED technically doesn't.

this is an interesting point

**Comment by lmolkova** [comment]
when sanitizing sensitive information, would we rather redact the values (i.e. `foo=REDACTED&bar=REDACTED`) or drop the attribute completely?

My preference is to preserve parameter names, order, etc rather than dropping the whole attribute.

I assume if users write sanitization, they do it as a processor (now) and will take the query attribute, parse, and sanitize individual properties.  If we provide a tracing configuration to sanitize arbitrary values, we'd do something similar.

TLDR: Opting-out needs configuration and this configuration would rather be a regex or a callback to sanitize than a boolean flag (drop/preserve).

Also, I feel we don't need to solve this problem prior to stability: it should not be breaking to switch between `recommended` vs `conditionall_required: when available` at any moment.
